imagebuildah/build.go

```diff
@@ -241,6 +255,9 @@ func (b *Executor) volumeCacheSave() error {
 		if !os.IsNotExist(err) {
 			return errors.Wrapf(err, "error checking for cache of %q in %q", archivedPath, cacheFile)
 		}
+		if err := os.MkdirAll(archivedPath, 0755); err != nil {
```

Comment:
At this point we should be operating on a read-write layer, so I don't expect whether or not the original image is on a writable filesystem to matter.